### PR TITLE
add functional component support

### PR DESCRIPTION
**Pull request by Dolanf**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,4 +68,5 @@ The `location` object has the following attributes:
...
 | `mounted`      | `boolean` | `false` | If `true`, the `handleRouteChanged` method of the wrappedComponent will be called with `(null, currentLocation)`when the component is mounted.                                                                                          |
-| `onlyPathname` | `boolean` | `true`  | If `true`, the `handleRouteChanged` method will only be called when the `pathname` of the location has been changed. If `false`, the changes of `search` or `hash` of the `location` will also trigger the `handleRouteChanged` method. |
\ No newline at end of file
+| `onlyPathname` | `boolean` | `true`  | If `true`, the `handleRouteChanged` method will only be called when the `pathname` of the location has been changed. If `false`, the changes of `search` or `hash` of the `location` will also trigger the `handleRouteChanged` method. |
+| `handleRouteChanged` | `function` |   | `handleRouteChanged` method is required when there is a functional component |
```
Comment (XiaocongDong) [inline README.md]: fix `there is` to `the decoratedComponent is`

File: src/index.js [hunk 2]
```diff
@@ -20,31 +21,49 @@ const onRouteChangedHOC = (DecoratedComponent, config = { mounted: false, onlyPa
       location: PropTypes.object.isRequired
     }
 
-    componentDidMount () {
-      if (typeof this.instanceRef.handleRouteChanged !== 'function') {
-        throw new Error(
-          'WrappedComponent lacks a handleRouteChanged(prevLocation, currLocation) for processing route changed event.'
-        )
+    __getHandleRouteChange = () => {
```
Comment (XiaocongDong) [inline src/index.js]: rename `__getHandleRouteChangedFunc`

**Review by XiaocongDong — CHANGES_REQUESTED** [review]
please fix

**Comment by Dolanf** [comment]
#1

**Pull request merged**